### PR TITLE
fix(auth): require secret for internal service mode

### PR DESCRIPTION
**Pull request by PranavaKCode**
## Description

- reads `MANTIS_INTERNAL_SERVICE_SECRET` into the SDK configuration
- sends `X-Internal-Secret` with the existing internal-service marker and user ID
- fails before any network request when internal-user mode lacks the required secret
- redacts the internal secret from debug transport logs
- documents and tests the authenticated internal-service contract

## Motivation and context

MantisAPI PR #2442 replaced the spoofable marker-only bypass with an authenticated `X-Internal-Secret` contract. SDK callers using `MANTIS_INTERNAL_USER_ID` must send that same provisioned secret or their requests fall through to browser authentication.

Fixes #11.

This is intentionally fail-closed: the SDK accepts a provisioned secret but never derives or logs it. Cookie authentication is unchanged.

## How has this been tested?

- `python -m ruff check .`
- `uv run --with pytest --with pytest-asyncio pytest -q` (71 tests passed)
- focused coverage verifies the valid three-header set, missing-secret failure, environment loading, and debug-log redaction
- `git diff --check`

## Screenshots (if appropriate):

Not applicable; this is an SDK authentication contract change.

## Types of changes

- ✅ Bug fix (non-breaking change which fixes an issue)
- ✅ Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project. **[REQUIRED]**
- [x] Ran `ruff check`. There are no style warnings. **[REQUIRED]**
- [x] Ran `pytest`. All new and existing tests passed. **[REQUIRED]**
- [ ] Recorded a video walkthrough showing the platform changes, testing used to verify the fix, and a brief review of the code changes made in this PR. **[REQUIRED]**
- [x] My change requires a change to the documentation and I have updated it accordingly.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: mantis_sdk/_http.py [hunk 1]
```diff
@@ -55,7 +56,12 @@ def auth_headers(self) -> dict[str, str]:
         if self.cookie:
             headers["cookie"] = self.cookie
         if self.config.internal_user_id:
+            if not self.config.internal_service_token:
+                raise ConfigurationError(
+                    "internal_service_token is required when internal_user_id is set"
+                )
             headers["X-Internal-Service"] = "true"
+            headers["X-Internal-Service-Token"] = self.config.internal_service_token
```
[the remark this answers is not in the document]
Comment (PranavaKCode) [reply mantis_sdk/_http.py]: Addressed in ca8bab2. The final backend contract now emits X-Internal-Secret rather than the superseded token header, mantis_sdk/transport.py redacts that exact header case-insensitively, and test_debug_log_redacts_internal_service_secret verifies the credential never appears in DEBUG logs.

**Review by FungousLand1941 — COMMENTED** [review]
I read through the auth/header changes and tests. The secret redaction and fail-closed internal-service path look solid. Two edge cases seem worth tightening before merge:

1. `MantisClient.__init__` still says that setting `internal_user_id` / `MANTIS_INTERNAL_USER_ID` is sufficient for backend-to-backend auth. With this PR that guidance is incomplete; the error should mention `internal_service_secret` / `MANTIS_INTERNAL_SERVICE_SECRET` too so the first configuration failure points to the full contract.

2. The PR says cookie authentication is unchanged, but `HttpClient.auth_headers()` now raises whenever `internal_user_id` is present without a secret, even if a valid cookie was supplied. That means a process using cookie auth with a stale/ambient `MANTIS_INTERNAL_USER_ID` will regress from working to raising `ConfigurationError`. If mixed cookie + internal configuration is intentionally fail-closed, I think it should be documented and covered by a test. If cookie should take precedence, the secret requirement probably needs to be gated to the internal-auth path.

The existing tests cover pure cookie and pure internal-service auth well; a mixed-auth regression test would make the intended precedence explicit.

**Comment by PranavaKCode** [comment]
Addressed the mixed-auth feedback in 74af9dc. A cookie now remains usable when an ambient internal user ID has no service secret; internal-only requests still fail closed. The README documents precedence when both complete methods are present, the constructor error names both required settings, and regression tests cover both mixed cases. Verified with Ruff, 74 passing tests, and git diff --check. Ready for a fresh review.

**Comment by PranavaKCode** [comment]
@MantisCartography Could you review the current head `74af9dcb` when available? The constructor guidance and both mixed cookie/internal-auth cases raised in the prior review are addressed, and Python 3.10-3.12 CI plus build are green. Please flag any remaining contract or deployment prerequisite before merge.

**Comment by PranavaKCode** [comment]
@FungousLand1941 Your two auth edge cases are addressed in head `74af9dcb`: the constructor error now names both required internal settings, and an ambient internal user ID no longer breaks cookie auth when the service secret is absent. README and mixed-auth regression tests cover the precedence, with all SDK CI checks green. If you have time for a fresh review, I would appreciate a look.

**Comment by PranavaKCode** [comment]
Deployment-contract note for review: current MantisAPI main derives `settings.INTERNAL_SERVICE_SECRET` from `SECRET_KEY` when backend env `INTERNAL_SERVICE_SECRET` is unset (the zero-config path from #2442). An external SDK process cannot obtain that derived value without sharing the Django `SECRET_KEY`, which it should not do. This SDK PR safely fails closed and its unit/CI coverage is green, but an actual external internal-user session needs an explicitly provisioned backend `INTERNAL_SERVICE_SECRET` and the matching SDK `MANTIS_INTERNAL_SERVICE_SECRET` deployed together (including blue/green backend instances), followed by an end-to-end auth check. Cookie-only callers do not need that credential. I have not verified production secret provisioning, so please do not treat the header tests alone as proof that external service mode works in production.
